src/libsyntax/parse/parser.rs

```diff
@@ -3877,20 +3877,29 @@ impl<'a> Parser<'a> {
                 err.cancel();
             }
             let seq_span = pat.span.to(self.prev_span);
-            let mut err = self.struct_span_err(comma_span,
-                                               "unexpected `,` in pattern");
+            let mut err = self.struct_span_err(
+                comma_span,
+                "unexpected `,` in pattern",
+            );
             if let Ok(seq_snippet) = self.sess.source_map().span_to_snippet(seq_span) {
                 err.span_suggestion(
                     seq_span,
-                    "try adding parentheses to match on a tuple..",
+                    &format!("try adding parentheses to match on a tuple{}", if is_match_arg {
+                        "..."
+                    } else {
+                        ""
+                    }),
                     format!("({})", seq_snippet),
                     Applicability::MachineApplicable
-                ).span_suggestion(
-                    seq_span,
-                    "..or a vertical bar to match on multiple alternatives",
-                    format!("{}", seq_snippet.replace(",", " |")),
-                    Applicability::MachineApplicable
                 );
+                if is_match_arg {
```

Comment:
Please leave a `FIXME(#54883)` here and above for undoing this once we have or-patterns.
cc https://github.com/rust-lang/rust/issues/54883 cc @dlrobertson

Reply:
My thinking is that this suggestion will _extremely rarely_ be correct in `let` statements. Anyone that want's to do `let x | y = foo();` will try _that_, whereas the typo `let x, y` is incredibly more likely to have meant a tuple as opposed to the (currently unsupported) `let x | y`.

Reply:
Hmm; I suppose you're right about that given that `a | b` won't type-check in any case.